services/ui-src/src/utils/other/mask.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Custom Masks Type Guard
+ * Add any future custom masks here!
+ */
+const customMasks = ["comma-separated"] as const;
+export type CustomMasks = typeof customMasks[number];
+export const isCustomMask = (x: any): x is CustomMasks =>
+  customMasks.includes(x);
+
+/**
+ * Checks if the provided string is contains numbers to mask
+ * @param {String} value
+ * @returns {String}
+ */
+export const isNumberStringMaskable = (value: string): boolean =>
+  value?.match(/\d/) ? true : false;
+
+/**
+ * Converts a number-like string to a comma seperated value
+ * @param {String} value
+ * @returns {String}
+ */
+export function convertToCommaSeparatedString(value: string): string {
```

Comment:
I think we should split this function in two since it is performing two tasks: 
1. Masking for thousands separator (for this app's purposes, should be run on every number input, including currency)
2. Masking for decimal places (should be run only on currency)

Reply:
Maybe even a third function that does the cleanup? Not necessarily.

Every number would need to go through: 
- Cleanup
- Comma-separation

Then optionally a number would additionally go through currency to get the decimals.

Reply:
<img width="848" alt="Screen Shot 2022-07-21 at 12 37 32" src="https://user-images.githubusercontent.com/27705928/180285089-78e224ba-8933-4448-a3d2-948890fe0e63.png">

Reply:
Kept the same for now since we decided on not writing the custom currency mask. Willing to split this up if you think it still needs to be though.